kernel/src/table_changes/log_replay.rs

```diff
@@ -344,15 +358,23 @@ impl RowVisitor for PreparePhaseVisitor<'_> {
                 }
             } else if getters[9].get_str(i, "cdc.path")?.is_some() {
                 *self.has_cdc_action = true;
-            } else if let Some(schema) = getters[10].get_str(i, "metaData.schemaString")? {
-                let configuration_map_opt = getters[11].get_opt(i, "metadata.configuration")?;
+            } else if let Some(id) = getters[10].get_str(i, "metaData.id")? {
```

Comment:
nit: If you do this:
```suggestion
            } else if let Some(id) = getters[10].get_opt(i, "metaData.id")? {
```
I believe type inference based on the `Metadata` constructor below will cause it to return `String` instead of `&str`.

Reply:
The whole thing would probably be simpler as:
```suggestion
            } else if let Some(id) = getters[10].get_opt(i, "metaData.id")? {
                let configuration_map_opt = getters[13].get_opt(i, "metaData.configuration")?;
                self.metadata = Some(Metadata {
                    id,
                    schema_string: getters[11].get(i, "metaData.schemaString")?,
                    partition_columns: getters[12].get(i, "metaData.partitionColumns")?,
                    configuration: configuration_map_opt.unwrap_or_else(HashMap::new),
                    ..Default::default() // Other fields are ignored
                });
```
(most of those single-use val actually make their lines longer than just passing the arg directly)